spec/models/user_spec.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe '#name_or_email' do
+    let!(:user) { User.new(id: 1, email: 'alice@example.com', name: '') }
```

Comment:
```suggestion
    let(:user) { User.new(email: 'alice@example.com', name: '') }
```

idはRailsが自動的に割り振るので固定値を使わないでください。
https://qiita.com/jnchito/items/6beb97e73934eb3e4dce

また、ここは普通のletでも問題ないですね